src/domain/networking/packets/EntityData.ts

```diff
@@ -9,31 +9,2064 @@
+                } else {
+                    entityData = new DataView(entityMessage.buffer.slice(entityMessage.byteOffset + entityMessagePosition));
+                }
+
+                let bytesRead = 0;
+                // WEBRTC TODO: This won't throw once all entity types are supported.
+                try {
+                    const parsedData = this.#readBitstreamToTree(entityData);
+                    bytesRead = parsedData.bytesRead;
+                    entityDataDetails = [...entityDataDetails, ...parsedData.entitiesDataDetails];
+                } catch (err) {
+                    // Discard the whole packet.
+                    return [];
+                }
+
+                assert(sectionLength !== bytesRead);
+                dataPosition += sectionLength;
+            }
+            dataPosition += entityMessagePosition;
+        }
+
+        // WEBRTC TODO: Address further C++ code.
+
+        /* eslint-enable @typescript-eslint/no-magic-numbers */
+
+        return entityDataDetails;
+    }
+
+    #readBitstreamToTree(data: DataView): ParsedData {
+        // C++ void Octree::readBitstreamToTree(const unsigned char * bitstream, uint64_t bufferSizeBytes,
+        //     ReadBitstreamToTreeParams& args)
+
+        let dataPosition = 0;
+
+        let entitiesDataDetails: EntityDataDetails[] = [];
+
+        while (dataPosition < data.byteLength) {
+            const numberOfThreeBitSectionsInStream
+            = this.#numberOfThreeBitSectionsInCode(data, dataPosition, data.byteLength);
+
+            const octalCodeBytes = this.#bytesRequiredForCodeLength(numberOfThreeBitSectionsInStream);
+            dataPosition += octalCodeBytes;
+
+            const parsedData = this.#readElementData(data, dataPosition);
+            dataPosition += parsedData.bytesRead;
+
+            entitiesDataDetails = [...entitiesDataDetails, ...parsedData.entitiesDataDetails];
+        }
+        return {
+            bytesRead: dataPosition,
+            entitiesDataDetails
+        };
+    }
+
+    #readElementData(data: DataView, position: number): ParsedData {
+        // C++ int Octree::readElementData(const OctreeElementPointer& destinationElement, const unsigned char* nodeData,
+        //     int bytesAvailable, ReadBitstreamToTreeParams& args) {
+
+        let dataPosition = position;
+
+        // 1 represents sizeof(unsigned char) in the C++ code.
+        if (data.byteLength - dataPosition < 1) {
+            console.error("Not enough meaningful data.");
+            return {
+                bytesRead: dataPosition - position,
+                entitiesDataDetails: []
+            };
+        }
+
+        // WEBRTC TODO: Read and use the variable colorInPacketMask.
+        dataPosition += 1;
+
+        // WEBRTC TODO: Do not hardcode value.
+        // eslint-disable-next-line
+                // @ts-ignore
+        const bytesForMask = 2;
+        dataPosition += bytesForMask;
+
+        // WEBRTC TODO: Address further C++ code.
```

Comment:
I think that this code may come into play when either (a) there are more entities in a domain than just one or two and they're store in octree nodes, or (b) entities are parented to others. In either case, there is some recursive code in the C++ that reads the entity data which we're probably going to need to implement ASAP.
This can perhaps be handled as a separate task (https://github.com/vircadia/vircadia-web-sdk/issues/131) - e.g., in parallel to reading the second entity type (https://github.com/vircadia/vircadia-web-sdk/issues/130).